app/graphql/resolvers/concerns/collection.rb

```diff
@@ -7,21 +7,126 @@ module Collection
         type ["Types::#{self::MODEL_CLASS}".safe_constantize], null: false
 
         argument :search, String, 'Search query', required: false
-        argument :order_by, String, 'Order by this field', required: false
-        argument :order, String, 'Order direction', required: false
+        argument :sort_by, String, 'Sort by this searchable field', required: false
+        argument :sort_direction, Types::SortDirectionEnum, 'Sort direction', required: false
+
+        if has_taxonomix?
+          argument :location, String, required: false
+          argument :location_id, String, required: false
+          argument :organization, String, required: false
+          argument :organization_id, String, required: false
+        end
+
+        delegate :has_taxonomix?, to: :class
+      end
+
+      def resolve(**kwargs)
+        filters = filter_list(**kwargs)
+
+        base_scope = authorized_scope.all
+        filters.reduce(base_scope) { |scope, filter| filter.call(scope) }
+      end
+
+      private
+
+      def user
+        context[:current_user]
+      end
+
+      def model_class
+        self.class::MODEL_CLASS
+      end
+
+      def authorized_scope
+        return model_class unless model_class.respond_to?(:authorized)
+
+        permission = model_class.find_permission_name(:view)
+        model_class.authorized_as(user, permission, model_class)
+      end
+
+      def search_options(search:, sort_by:, sort_direction:)
```

Comment:
I wonder, could we perhaps reuse some logic from the api base controllers here?

Reply:
We just have this line that we could reuse. I think it's not worth sharing. But if you insist, I'll extract this some kind of shared concern.
https://github.com/theforeman/foreman/blob/7500537eefac12d1e58daa58c1efd346c6dc719c/app/controllers/api/base_controller.rb#L216-L218

Reply:
i don't insist, just wondering because we have quite similar logic for search filtering in the api and wanted to avoid duplication if it makes sense.